rcp/org.testeditor.rcp4.tcltestrun/src/org/testeditor/rcp4/tcltestrun/MavenExecutor.java

```diff
@@ -0,0 +1,142 @@
+	 * 
+	 */
+	public int execute(String goal, String pathToPom) {
+		System.setProperty("maven.multiModuleProjectDirectory", pathToPom);
+		MavenCli cli = new MavenCli();
+		int result = cli.doMain(new String[] { "clean", goal }, pathToPom, System.out, System.err);
+		return result;
+	}
+
+	/**
+	 * Executes a maven build in a new jvm. The executable of the current jvm is
+	 * used to create a new jvm.
+	 * 
+	 * @param goal
+	 *            of maven to be exeucted.
+	 * @param pathtoPom
+	 *            path to the folder where the pom.xml is located.
+	 * @param testParam
+	 *            pvm parameter to identify the test case to be executed.
+	 * @return int with exit code
+	 * @throws IOException
+	 *             on failure
+	 */
+	public int executeInNewJvm(String goal, String pathtoPom, String testParam) throws IOException {
+		int result = 1; // unspecified error code != 0
+		String jvm = System.getProperty("java.home") + File.separator + "bin" + File.separator + "java";
+		List<String> command = new ArrayList<String>();
+		command.add(jvm);
+		command.add("-cp");
+		command.add(getClassPath());
+		command.add(this.getClass().getName());
+		command.add(goal);
+		command.add(pathtoPom);
+		command.add(testParam);
+		logger.trace("Execute maven in new jvm with: {}", command);
+		ProcessBuilder processBuilder = new ProcessBuilder();
+		processBuilder.inheritIO();
+		processBuilder.redirectErrorStream(true);
+		processBuilder.command(command);
+		Process process = processBuilder.start();
+		try {
+			result = process.waitFor();
+		} catch (InterruptedException e) {
+			e.printStackTrace();
+		}
+		return result;
+	}
+
+	/**
+	 * Builds the classpath to maven embedder libs, this class and the
+	 * dependencies.
+	 * 
+	 * @return String with the classpath
+	 */
+	private String getClassPath() {
+		List<String> cp = new ArrayList<String>();
+		cp.addAll(Bundles.getClasspathEntries(Bundles
+				.findDependencyBundle(MavenPluginActivator.getDefault().getBundle(), "org.eclipse.m2e.maven.runtime")));
+		Bundle bundle = FrameworkUtil.getBundle(this.getClass());
+		cp.addAll(Bundles.getClasspathEntries(bundle));
+		Bundle[] bundles = bundle.getBundleContext().getBundles();
+		for (String sname : new String[] { "org.slf4j.api", "org.eclipse.m2e.maven.runtime.slf4j.simple",
+				"javax.inject" }) {
+			Bundle dependency = Bundles.findDependencyBundle(Bundles.findDependencyBundle(
+					MavenPluginActivator.getDefault().getBundle(), "org.eclipse.m2e.maven.runtime"), sname);
+			if (dependency != null) {
+				cp.addAll(Bundles.getClasspathEntries(dependency));
+			}
+		}
+		StringBuffer sb = new StringBuffer();
+		sb.append("\"");
+		for (String string : cp) {
+			sb.append(string).append(File.pathSeparator);
+		}
+		sb.append("\"");
+		return sb.toString();
+	}
+
+	/**
+	 * Entry point of the new jvm used for the maven build.
+	 * 
+	 * @param args
+	 *            the maven parameter.
+	 */
+	public static void main(String[] args) {
```

Comment:
The  main method is indirectly called with the method executeInNewJvm. That method creates  a new jvm for the maven build.
This is necessary to execute tycho tests. (Exquinox can't be run in an equinox)
It also protects the TE for problems in the fixtures.